### PR TITLE
Support Narrow for 1D input data

### PR DESCRIPTION
**Pull request by Deliadd**
@zhichao-li

### CODE REVIEW

**Comment by Deliadd** [comment]
local scala test has passed

**Comment by Deliadd** [comment]
Jenkins passed 
Check the link: http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Validation/1957/

File: zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/NarrowSpec.scala [hunk 1]
```diff
@@ -20,13 +20,26 @@ import com.intel.analytics.bigdl.nn.{Narrow => BNarrow}
...
+import com.intel.analytics.zoo.pipeline.api.keras.models.{Model, Sequential}
 
 import scala.util.Random
 
 class NarrowSpec extends ZooSpecHelper {
 
+  "Narrow Zoo 1D" should "be the same as BigDL" in {
+    val blayer = BNarrow[Float](1, 1, 2)
+    val input1 = Parameter[Float](inputShape = Shape(4), name = "input1")
+    val zlayer = new ZNarrow[Float](1, 1, 2).from(input1)
```
Comment (zhichao-li) [inline zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/NarrowSpec.scala]: ZNarrow(0, 0, 2)?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/Narrow.scala [hunk 2]
```diff
@@ -85,7 +88,8 @@ class Narrow[T: ClassTag](
 
   override def doBuild(inputShape: Shape): AbstractModule[Tensor[T], Tensor[T], T] = {
     val (positiveDim, positiveLength) = getPositiveDimAndLength(inputShape)
-    val layer = com.intel.analytics.bigdl.nn.Narrow(positiveDim + 1, offset + 1, positiveLength)
+    val offset2 = if (inputShape.toSingle().toArray.length>1) offset + 1 else offset
+    val layer = com.intel.analytics.bigdl.nn.Narrow(positiveDim + 1, offset2, positiveLength)
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/Narrow.scala]: Why do we need to distinguish these two things?

**Comment by Deliadd** [comment]
Error with python unittest ：java.lang.UnsupportedOperationException: empty.reduceLeft

**Comment by zhichao-li** [comment]
jenkins?

**Comment by Deliadd** [comment]
Jenkins passed. ID is 2004
Check the link here: http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Validation/2004/

**Pull request merged**